theseus_gui/src/store/search.js

```diff
@@ -0,0 +1,183 @@
+import { defineStore } from 'pinia'
+import generated from '@/generated'
+
+export const useSearch = defineStore('searchStore', {
+  state: () => ({
+    searchResults: [],
+    searchInput: '',
+    totalHits: 0,
+    currentPage: 1,
+    pageCount: 1,
+    offset: 0,
+    filter: 'Relevance',
+    categories: {},
+    loaders: {},
+    environments: {
+      client: false,
+      server: false,
+    },
+    activeVersions: [],
+    openSource: false,
+    limit: 20,
+  }),
+  actions: {
+    initFacets() {
```

Comment:
facets shouldn't have to be "initialized" IMO. the way the frontend does is that straight from vue it uses the generated/tag file to generate the search filters. the selected categories and loaders are just an array which are pushed and removed too. 

I think that approach would simplify some things you have going on here

Reply:
Initialized probably isn't the best name for this method - it's really just pulling the modpack categories and loaders out and pushing them into the arrays in the store. Will rename.

Reply:
My point here is that the method isn't needed, you can do it directly in the Vue component which is a lot cleaner IMO

Reply:
The actual generation of the facets doesn't really have anything to do with the presentation (ie. component) either, just reading them out into the SearchFilters. I think it makes more sense to have that data management in the store. The Browse page shouldn't necessarily care what generating facets is supposed to look like.